galley/pkg/kube/converter/converter.go

```diff
@@ -51,6 +54,8 @@ var converters = map[string]Fn{
 	"legacy-mixer-resource": legacyMixerResource,
 	"auth-policy-resource":  authPolicyResource,
 	"kube-ingress-resource": kubeIngressResource,
+	// TODO: Disable unless a command-line flag is set?
+	"kube-service-resource": kubeServiceResource,
```

Comment:
Thanks for picking this up. I was about to start this.

The feature is not disabled here. The enabling/disabling is done through metadata model which drives code-gen (see galley/tools). I am planning to do some work in this area to make the metadata model configurable through external mechanisms. That maybe the ideal way to control this at the limit.

Reply:
I see ServiceEntry is already defined in galley/tools/gen-meta/metadata.yaml.

What else does Peter need to do?

Reply:
That one is for reading service entries from K8s. He needs to add an entry for the services, similar to the Ingress one.

Also we need to protect this behind a flag or environment variable, so that we don't pollute perf current numbers and have the ability to see the impact of this on perf. We will most likely need to further enhancement in the rest of Galley to ensure that the chattiness here will not cause excessive churn.

Reply:
Now behind ISTIO_CONVERT_K8S_SERVICE env var in source.go (no idea what a good var name would be though).